packages/shared_preferences/shared_preferences_tool/lib/main.dart

```diff
@@ -17,22 +18,47 @@ class _SharedPreferencesTool extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return DevToolsExtension(
-      child: Builder(builder: (BuildContext context) {
-        return FutureBuilder<Object>(
-            future: serviceManager.onServiceAvailable,
-            builder: (BuildContext context, AsyncSnapshot<Object> snapshot) {
-              if (snapshot.connectionState == ConnectionState.waiting) {
-                return const Center(
-                  child: CircularProgressIndicator(),
-                );
-              }
-
-              return const SharedPreferencesStateProvider(
-                child: SharedPreferencesBody(),
-              );
-            });
-      }),
+    return const DevToolsExtension(
+      child: _ConnectionManager(),
     );
   }
 }
+
+class _ConnectionManager extends StatefulWidget {
+  const _ConnectionManager();
+
+  @override
+  State<_ConnectionManager> createState() => _ConnectionManagerState();
+}
+
+class _ConnectionManagerState extends State<_ConnectionManager> {
+  @override
+  void initState() {
+    super.initState();
+    // Used to move the application back to the loading state on the simulated
+    // environment when the developer disconnects the app.
+    serviceManager.registerLifecycleCallback(
+      ServiceManagerLifecycle.afterCloseVmService,
+      (_) {
+        setState(() {});
+      },
+    );
```

Comment:
does this have any implications when _not_ using the simulated environment?

Reply:
gonna double check

Reply:
Just tested it again and we have no implications when running on the actual DevTools.